Centaurus.Domain/Snapshot/DALModelExtensions/EffectModelExtensions.cs

```diff
@@ -3,23 +3,32 @@
 using Centaurus.Xdr;
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Text;
 
 namespace Centaurus.Domain
 {
     public static class EffectModelExtensions
     {
-        public static EffectModel FromEffect(Effect effect, int index)
+        public static EffectModel FromEffect(this Effect effect, int index)
         {
             return new EffectModel
             {
-                Index = index,
-                Apex = (long)effect.Apex,
+                Id = BitConverter.GetBytes(effect.Apex).Concat(BitConverter.GetBytes(index)).ToArray(),
```

Comment:
Instead of encoding on the fly, extract a shared `EffectModel` extension method (or put it somewhere else – up to you). Also implement the decoding method there.

```c#
public byte[] EncodeEffectId(long apex, int index) {
  var buffer = new byte[12];
  var span = buffer.AsSpan();
  BinaryPrimitives.WriteInt64LittleEndian(buffer.Slice(0, 8), apex);
  BinaryPrimitives.WriteInt64LittleEndian(buffer.Slice(8, 4),index);
  return buffer;
}
```

```